File: api/src/main/java/org/delivery/api/common/error/UserErrorCode.java
package org.delivery.api.common.error;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.http.HttpStatus;

@AllArgsConstructor
@Getter
public enum UserErrorCode implements ErrorCodeIfs{
    USER_NOT_FOUND(HttpStatus.BAD_REQUEST.value(), 400, "Wrong request")
    ;

    private final Integer httpStatusCode;

    private final Integer errorCode;

    private final String description;
    ;
}

File: common/src/main/kotlin/org/delivery/common/error/UserErrorCode.kt
package org.delivery.common.error

enum class UserErrorCode(
    private val httpStatusCode: Int,
    private val errorCode: Int,
    private val description: String,
) : ErrorCodeIfs {

/*    OK(200, 200, "success"),
    BAD_REQUEST(400, 500, "bad request"),
    SERVER_ERROR(500, 500, "server error"),
    NULL_POINT(500, 512, "null point"),*/
    USER_NOT_FOUND(400, 400, "사용자를 찾을 수 없음.")

    ;

    override fun getHttpStatusCode(): Int {
        return this.httpStatusCode
    }

    override fun getErrorCode(): Int {
        return this.errorCode
    }

    override fun getDescription(): String {
        return this.description
    }


}